app/Dao/UserDao.php

```diff
@@ -108,4 +99,33 @@ public function deleteUserById($id): void
         $user = User::findOrFail($id);
         $user->delete();
     }
+
+    /**
+     * Login user by email and password
+     * @param Request data
+     * @return User
+     */
+    public function checkLogin($user): ?object
```

Comment:
Return type will be` null|User`.
```suggestion
    public function checkLogin($user): null|User
```